docs/src/model_creation/dsl_advanced.md

```diff
@@ -272,6 +272,39 @@ sol = solve(oprob)
 plot(sol)
 ```
 
+### [Turning off species, parameter, and variable inferring](@id dsl_advanced_options_require_dec)
+In some cases it may be desirable for Catalyst to not infer species and parameters from the DSL, as in the case of reaction networks with very many variables, or as a sanity check that variable names are written correctly. To turn off inferring, simply add the `@require_declaration` macro to one of the lines of the `@reaction_network` declaration. Having this macro means that every single variable, species, or parameter will have to be explicitly declared using the `@variable`, `@species`, or `@parameter` macro. In the case that the DSL parser encounters an undeclared symbolic, it will error with an `UndeclaredSymbolicError` and print the reaction or equation that the undeclared symbolic was found in. 
```

Comment:
"To turn off inferring, simply use the `@require_declaration` option when using the `@reaction_network` DSL. This will require every variable, species, or parameter used within the DSL to be explicitly declared using the `@variable`, `@species`, or `@parameter` options."

Usage of "@..." within the DSL are not macro, and really don't have anything with them to do. It is simply the notation we have opted to use within the DSL to designate options (i.e. every input to the macro that is not a reaction line)